src/pudl/analysis/ferc1_eia.py

```diff
@@ -431,596 +436,139 @@ def get_features(self, clobber=False):
-        candiate matches in ``features_all``. But we only want one match for each FERC1
-        plant record. So this method uses the coeficients from the model (which are a
-        measure of the importance of each of the features/columns in ``features_all``)
-        so weight the feature vecotrs. With the sum of the weighted feature vectors for
-        each model match, this method selects the hightest scoring match via
-        :meth:`calc_best_matches()`.
+def remove_murky_matches(match_df, difference_threshold):
+    """Remove matches that are not clear winners."""
+    match_df = (
+        match_df.reset_index()
+        .merge(
+            pudl.helpers.count_records(match_df, "record_id_ferc1", "count"), how="left"
+        )
+        .sort_values(by=["record_id_ferc1", "prob_of_match"])
+    )
+    match_df["diff"] = match_df["prob_of_match"].diff()
+    match_df = match_df.groupby("record_id_ferc1").tail(1)
+    match_df = match_df[
+        (match_df["count"] == 1) | match_df["diff"] >= difference_threshold
+    ]
+    return match_df
 
-        Args:
-            features_train (pandas.DataFrame): feature vectors between training
-                data from FERC plants and EIA plant-parts. Result of
-                :meth:`Features.make_features()`.
-            features_all (pandas.DataFrame): feature vectors between all data
-                from FERC plants and EIA plant-parts. Result of
-                :meth:`Features.make_features()`.
 
-        Returns:
-            pandas.DataFrame: the best matches between ferc1 plant records and
-            the EIA plant-parts. Each ferc1 plant record has a maximum of one
-            best match. The dataframe has a MultiIndex with `record_id_eia` and
-            `record_id_ferc1`.
-        """
-        # actually run a model using the "best" model!!
-        logger.info("Fit and predict a model w/ the highest scoring hyperparameters.")
-        # this returns all matches that the model deems good enough from the
-        # candidate matches in the `features_all`
-        matches_model = self.fit_predict_lrc(
-            best=self.best,
-            features_train=features_train,
-            features_all=features_all,
-            train_df=self.train_df,
-        )
-        # weight the features of model matches with the coeficients
-        # we need a metric of how different each match is
-        # merge in the IRQ of the full options
-        self.matches_model = pd.merge(
-            self.calc_match_stats(matches_model),
-            self.calc_match_stats(features_all)[["iqr"]],
-            left_index=True,
-            right_index=True,
-            how="left",
-            suffixes=("", "_all"),
-        )
-        matches_best_df = self.calc_best_matches(self.matches_model, 0.02).pipe(
-            self.override_best_match_with_training_df, self.train_df
-        )
-        return matches_best_df
+def overwrite_bad_predictions(match_df, train_df):
```

Comment:
This function does the same thing as `MatchManager.override_best_match_with_training_df()`. It's pretty much the same, I think I just cleaned up the logic on the merge. Also I changed "override" to "overwrite" lol but happy to go back. That's definitely a nit.

Reply:
ha! i love that i've been writing over*ride* for literal years and did not notice that this was wrong.